src/components/Project/ProjectPage.tsx

```diff
@@ -1,15 +1,73 @@
 "use client";
 
-import { useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 import ProjectTabs from "../ProjectTabs";
-import { ProjectTab } from "@/types/const";
+import { ProjectIDType, ProjectTab } from "@/types/const";
 import ProjectDetailsComponent from "../ProjectMetaDataTable/ProjectDetailsComponent";
 import ContractDetailsPage from "../ProjectContract/ContractDetailsPage";
 import ProjectDistribution from "./ProjectDistribution";
 import MockData from "./project.json";
+import { useAccount } from "wagmi";
+import getDecodedAttestationData from "@/lib/eas/getDecodedAttestationData";
+import { useProjectProvider } from "@/context/ProjectProvider";
+import { useParams } from "next/navigation";
+import { usePaymasterProvider } from "@/context/Paymasters";
+import { getProjects } from "@/lib/eas/getProjects";
 
 const ProjectPage = () => {
   const [tabContent, setTabContent] = useState<ProjectTab>("project");
+  const [mockData, setMockData] = useState(null);
+  const { address } = useAccount();
+  const { setName, setDescription } = useProjectProvider();
+  const { id } = useParams<ProjectIDType>();
+  const { error }: any = usePaymasterProvider();
+  let refUid = "";
+  let uid = id;
+  let accountAddress = address;
+
+  const getAddress = (projects: any) => {
+    if (projects?.data.length > 0) {
+      let refAttestation = projects.data[projects.data.length - 1];
+      refUid = refAttestation.result[5];
+      if (refUid === id) {
+        uid = refAttestation.result[0];
+        accountAddress = refAttestation.result[7];
+      }
+    }
+  };
+
+  const fetchAttestation = useCallback(async () => {
+    const queryParam = address ? `?address=${address}` : "";
+    const projects: any = await getProjects(queryParam);
+    getAddress(projects);
+    const response = await fetch(
+      `/api/attestation?address=${accountAddress}&uid=${uid}`
+    );
+    if (!response.ok) return false;
+    const data = await response.json();
+    const mapped = data.data.map((attestation: any) =>
+      getDecodedAttestationData(attestation.result)
+    );
+    const extractedData: any = {};
+    mapped.forEach((subArray: any) => {
+      subArray.forEach((item: any) => {
+        const key = item.name === "metadataUri" ? "description" : item.name;
+        extractedData[key] = item?.value?.value;
+      });
+    });
+    let dashboardData: any = MockData;
+    dashboardData["name"] = extractedData["title"];
+    dashboardData["description"] = extractedData["description"];
+    setMockData(dashboardData);
+    setName(dashboardData["name"]);
+    setDescription(dashboardData["description"]);
+    return extractedData;
+  }, []);
```

Comment:
useCallback is not needed if you move `fetchAttestation` to a lib.

please keep React components minimal in size

Reply:
please create a new hook: `useAttestation`